### ISSUE TITLE
Updating OCaml version

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/ocamhaxe#13]
We're currently defaulting to OCaml 4.09.0, which locks us to lcaml-lsp-server 1.4.1 from March 2021. The current version of that is 1.17.0 which has an `ocaml >= 4.14`. However, we're also locked to camlp5 8.00 which has `ocaml < 4.13.0`, meaning `4.12.1`. This would allow us at most ocaml-lsp-server 1.8.3 from October 2021, which doesn't seem like much of an upgrade.

I just noticed that camlp5 8.00.03 is the last version that has [the patch](https://github.com/ocaml-opam/opam-repository-mingw/blob/sunset/packages/camlp5/camlp5.8.00.03/files/camlp5-rel8.00.03.patch) on the sunset repository, and coincidentally it's the first version that has `ocaml < 4.15.0`, which _should_ allow us an upgrade to 4.14.

I'll see if that works... The reason I'm opening this issue is mostly to rant, but also because my IDE currently looks like this:

<img width="621" alt="Code_NqJSvOZZP6" src="https://github.com/HaxeFoundation/ocamhaxe/assets/634365/e4426c0e-8e69-4896-a97e-afda174b0f05">

I don't remember what versions I had before, so I'm trying to recreate to workable working environment.

**Comment by Simn** [comment]
Works about as well as I was expecting:

```
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
∗ installed base-bigarray.base
∗ installed base-threads.base
∗ installed base-unix.base
∗ installed ocaml-variants.4.14.1+mingw64
∗ installed ocaml-config.2
∗ installed ocaml.4.14.1
Done.
# Run eval $(opam env --root=C:/git/ocamhaxe/.opam) to update the current shell environment
Fatal error:
Not_found
Backtrace:
  Raised at Stdlib__list.assq in file "list.ml" (inlined), line 199, characters 10-25
  Called from Re__Core.final in file "lib/core.ml", line 223, characters 4-26
  Called from OpamStubs in file "src/stubs/opamWin32Stubs.ml" (inlined), line 20, characters 0-60
  Called from OpamConsole.enable_win32_vt100.restore_console in file "src/core/opamConsole.ml", line 265, characters 12-51
  Called from Stdlib.at_exit.(fun) in file "stdlib.ml", line 553, characters 62-65
  Called from Stdlib.do_at_exit in file "stdlib.ml" (inlined), line 556, characters 20-39
  Called from Stdlib.exit in file "stdlib.ml", line 559, characters 2-15
  Called from OpamMain.main_catch_all in file "src/client/opamMain.ml" (inlined), line 188, characters 6-10
  Called from OpamStd.Op.(@@) in file "src/core/opamStd.ml" (inlined), line 1185, characters 17-20
  Called from OpamMain.run in file "src/client/opamMain.ml", line 267, characters 2-695
```

**Comment by Simn** [comment]
My IDE issues came from something rather unexpected which I reported here: https://github.com/ocaml/ocaml-lsp/issues/1224

**Comment by Simn** [comment]
Dune 3.16 currently doesn't work on Windows with OCaml 4.09 and they're talking about dropping support for it here: https://github.com/ocaml/dune/issues/10799#issuecomment-2270061283

This might make it necessary to look into #16 soon, unless we want to pin an older dune version which doesn't seem ideal. At the moment I'm running into the dreaded camlp5 `The -prefix directory must be absolute.` problem again and according to [this comment](https://github.com/ocaml-opam/opam-repository-mingw/issues/14#issuecomment-2210981735) the required patches aren't going to be migrated either. I saw that @tobil4sk is trying to deal with that [here](https://github.com/camlp5/camlp5/pull/108), but it likely will be a while before that's released in some form.

So in summary I'm not sure how to proceed right now.

**Comment by tobil4sk** [comment]
For now, we can maybe fork the `opam-repository-mingw` repository to port the camlp5 patch to a newer version ourselves?

**Comment by Simn** [comment]
That indeed seems like the most straightforward solution, though I'm worried about having to keep that up to date. Ideally such a repository would only contain what we're interested in and not potentially outdated versions of everything else.
